File: services/student-service/src/main/java/com/infosys/studentservice/models/Student.java
package com.infosys.studentservice.models;

import lombok.Data;
import lombok.NoArgsConstructor;

import javax.persistence.*;
import java.util.List;

@Entity
@Table(name = "student")
@Data @NoArgsConstructor
public class Student {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Integer id;
    private String name;
    private String schoolid;
    private String email;
    private String phone;
    private Integer ssn;
    private String street;
    private String city;
    private String state;
    private Integer zipcode;

    @Transient
    private School school;

    public Student(String name, Integer id) {
        this.name = name;
        this.id = id;
    }

}

File: frontend/src/app/models/student.ts
import {School} from './school'

export interface Student {
    name: string;
    id?: Number;
    school?: School;
    email: string;
    ssn: Number
    phone: string
    street: string;
    city: string;
    state: string;
    zipcode: Number;
    schoolid?: Number;
}